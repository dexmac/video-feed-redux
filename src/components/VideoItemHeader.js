import React from "react";
import VideoItemTitle from "./VideoItemTitle";
import VideoItemViewsNumber from "./VideoItemViewsNumber";
import PropTypes from "prop-types";

const VideoItemHeader = ({ title = "", viewsNumber }) => (
  <div className="video-item-header">
    <VideoItemTitle className="video-item-title" title={title} />
    <VideoItemViewsNumber
      className="video-item-views-number"
      viewsNumber={viewsNumber}
    />
  </div>
);

VideoItemHeader.propTypes = {
  title: PropTypes.string,
  viewsNumber: PropTypes.number
};

export default VideoItemHeader;
