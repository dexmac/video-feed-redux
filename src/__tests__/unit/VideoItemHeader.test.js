import React from "react";
import VideoItemHeader from "../../components/VideoItemHeader";

describe("VideoItemHeader", () => {
  let props;
  let mountedVideoItemHeader;

  /**
   * Clearing props before each test
   */
  beforeEach(() => {
    props = {
      title: undefined,
      viewsNumber: undefined
    };
    mountedVideoItemHeader = undefined;
  });

  /**
   * Shallow rendering vs. snapshot test
   */
  it("renders correctly && matches snapshot", () => {
    const tree = shallow(<VideoItemHeader />);
    expect(tree).toMatchSnapshot();
  });

  /**
   * Mounts the component using Enzyme if it is not mounted yet
   * @returns {object} - returns the mounted component
   */
  const videoItemHeader = () => {
    if (!mountedVideoItemHeader) {
      mountedVideoItemHeader = mount(<VideoItemHeader {...props} />);
    }
    return mountedVideoItemHeader;
  };

  describe("when `title` && 'views' are defined", () => {
    beforeEach(() => {
      props.viewsNumber = 12345;
      props.title = "myTitle";
    });

    it("sets the rendered child component props to the same values", () => {
      const viewsNumber = videoItemHeader().find("VideoItemViewsNumber");
      expect(viewsNumber.props().viewsNumber).toEqual(12345);

      const title = videoItemHeader().find("VideoItemTitle");
      expect(title.props().title).toEqual("myTitle");
    });
  });

  describe("when `title` && 'views' are undefined", () => {
    beforeEach(() => {
      props.viewsNumber = undefined;
      props.title = undefined;
    });

    it("sets the rendered child component props to empty string", () => {
      const viewsNumber = videoItemHeader().find("VideoItemViewsNumber");
      expect(viewsNumber.props().viewsNumber).toEqual(undefined);

      const title = videoItemHeader().find("VideoItemTitle");
      expect(title.props().title).toEqual("");
    });
  });
});
