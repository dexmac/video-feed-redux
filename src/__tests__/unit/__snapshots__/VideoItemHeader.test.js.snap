// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItemHeader renders correctly && matches snapshot 1`] = `
<div
  className="video-item-header"
>
  <VideoItemTitle
    className="video-item-title"
    title=""
  />
  <VideoItemViewsNumber
    className="video-item-views-number"
  />
</div>
`;
